lib/improver/utilities/temporal_interpolation.py

```diff
@@ -0,0 +1,419 @@
+
+import iris
+from iris.exceptions import CoordinateNotFoundError
+
+from improver.utilities.temporal import iris_time_to_datetime
+from improver.utilities.solar import DayNightMask, calc_solar_elevation
+from improver.utilities.cube_manipulation import merge_cubes
+from improver.utilities.cube_checker import check_coord_datatypes
+from improver.utilities.spatial import (
+    lat_lon_determine, transform_grid_to_lat_lon)
+
+
+class TemporalInterpolation(object):
+
+    """
+    Interpolate data to intermediate times between the validity times of two
+    cubes. This can be used to fill in missing data (e.g. for radar fields) or
+    to ensure data is available at the required intervals when model data is
+    not available at these times.
+    """
+
+    def __init__(self, interval_in_minutes=None, times=None,
+                 interpolation_method='linear'):
+        """
+        Initialise class.
+
+        Keyword Args:
+            interval_in_minutes (int):
+                Specifies the interval in minutes at which to interpolate
+                between the two input cubes. A number of minutes which does not
+                divide up the interval equally will raise an exception::
+                    e.g. cube_t0 valid at 03Z, cube_t1 valid at 06Z,
+                    interval_in_minutes = 60 --> interpolate to 04Z and 05Z.
+            times (list or tuple of datetime.datetime objects):
+                A list of datetime objects specifying the times to which to
+                interpolate.
+            interpolation_method (str):
+                Method of interpolation to use. Default is linear.
+                Only methods in known_interpolation_methods can be used.
+
+        Raises:
+            ValueError: If neither interval_in_minutes nor times are set.
+            ValueError: If interpolation method not in known list.
+
+        """
+        if interval_in_minutes is None and times is None:
+            raise ValueError("TemporalInterpolation: One of "
+                             "'interval_in_minutes' or 'times' must be set. "
+                             "Currently both are none.")
+
+        self.interval_in_minutes = interval_in_minutes
+        self.times = times
+        known_interpolation_methods = ['linear', 'solar', 'daynight']
+        if interpolation_method not in known_interpolation_methods:
+            raise ValueError("TemporalInterpolation: Unknown interpolation "
+                             "method {}. ".format(interpolation_method))
+        self.interpolation_method = interpolation_method
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<TemporalInterpolation: interval_in_minutes: {}, '
+                  'times: {}, method: {}>')
+        return result.format(self.interval_in_minutes, self.times,
+                             self.interpolation_method)
+
+    def construct_time_list(self, initial_time, final_time):
```

Comment:
Function unchanged

Reply:
What does that mean?

Reply:
Sorry - this was just to help me keep track of what needs reviewing and what was moved without changes.